scm-plugins/scm-git-plugin/src/main/java/sonia/scm/repository/spi/GitPullCommand.java

```diff
@@ -203,41 +182,49 @@ private PullResponse pullFromScmRepository(Repository sourceRepository)
 
     org.eclipse.jgit.lib.Repository source = null;
 
-    try
-    {
+    try {
       source = Git.open(sourceDirectory).getRepository();
       response = new PullResponse(push(source, getRemoteUrl(targetDirectory)));
-    }
-    finally
-    {
+    } finally {
       GitUtil.close(source);
     }
 
     return response;
   }
 
-  private PullResponse pullFromUrl(URL url)
-    throws IOException
-  {
-    logger.debug("pull changes from {} to {}", url, repository.getId());
+  private PullResponse pullFromUrl(PullCommandRequest request)
+    throws IOException {
+    logger.debug("pull changes from {} to {}", request.getRemoteUrl(), repository.getId());
 
     PullResponse response;
     Git git = Git.wrap(open());
 
-    try
-    {
+    try {
       //J-
       FetchResult result = git.fetch()
+        .setCredentialsProvider(
+          new UsernamePasswordCredentialsProvider(
+            Strings.nullToEmpty(request.getUsername()),
+            Strings.nullToEmpty(request.getPassword())
```

Comment:
Well, it doesn't hurt. Before i only set this if both (username, pw) were provided by the request but it really shouldn't make a difference. The credentials is only used if it is needed and otherwise the provided credentials will be ignored.